src/lib.rs

```diff
@@ -6980,9 +6979,11 @@ mod tests {
 
         pipe.server.stream_recv(4, &mut buf).unwrap();
 
+        assert_eq!(pipe.server.streams.get(4).unwrap().recv.max_data_next(), 22);
+
         let frames = [frame::Frame::Stream {
```

Comment:
@ghedo Hi Alessandro, 

I've a question regarding some of the unit tests such as this one that sends frames to the server. In these tests, a frame is sent to the server, but it has no effect on the unit tests since `pipe.server.stream_recv` is not called afterwards.

I believe that the server receives the frame but do nothing with it. In this example:
In the master branch, the first frame adds 7 bytes to the `max_data_next value` and so the MAX_STREAM_DATA frame that is received has a max value of 22 (15+7). The one byte from the second frame does not affect the `max_data_next` value.
In my PR, I have to send 9 bytes in the first frame to cause the server sending a MAX_STREAM_DATA frame (`len > max_data - max_data_incr == 9 > 15-7)`. Sending 7 bytes in the first frame and 2 in the second does not cause MAX_STREAM_DATA frame. 

I've left the frames in this test and in the test on line 6864 because I'm not sure of the intent here. Just wanted to raise the point. ;)